### PR TITLE
Add check for minimal Fortran compiler versions

### PR DESCRIPTION
**Pull request by aradi**

### CODE REVIEW

File: cmake/DftbPlusUtils.cmake [hunk 1]
```diff
@@ -700,3 +683,27 @@ function (dftbp_create_lcov_targets lcov genhtml lcov_output_dir lcov_base_dir l
...
+#       message is printed about the check.
+#
+function (dftbp_check_minimal_compiler_version lang compiler_versions)
+  while(compiler_versions)
+    list(POP_FRONT compiler_versions compiler version)
+    if("${CMAKE_${lang}_COMPILER_ID}" STREQUAL "${compiler}")
+      if (CMAKE_${lang}_COMPILER_VERSION VERSION_LESS "${version}")
+        message(FATAL_ERROR "${compiler} ${lang} compiler is too old (found \"${CMAKE_${lang}_COMPILER_VERSION}\" < \"${version}\")")
+      else()
+        message(STATUS "${compiler} ${lang} compiler is recent enough (found \"${CMAKE_${lang}_COMPILER_VERSION}\" >= \"${version}\")")
```
Comment (awvwgk) [inline cmake/DftbPlusUtils.cmake:705]: Is this message relevant? CMake will print out the compiler version in any case.

File: cmake/DftbPlusUtils.cmake [hunk 2]
```diff
@@ -223,6 +223,10 @@ function (dftbp_ensure_config_consistency)
 
   endif()
 
+  # Check minimal compiler versions
+  set(fortran_minimal_versions "GNU;9.0" "Intel;19.0" "NAG;7.0")
```
Comment (awvwgk) [inline cmake/DftbPlusUtils.cmake:227]: Does this work reliably with CMake using the semicolon as default delimiter? Or does it just assume that we will later always remove the entries of the list in pairs?

File: cmake/DftbPlusUtils.cmake (hunk 1 above)
Comment (aradi) [reply cmake/DftbPlusUtils.cmake]: We can leave it away, indeed.

File: cmake/DftbPlusUtils.cmake (hunk 2 above)
Comment (aradi) [reply cmake/DftbPlusUtils.cmake:227]: Yes, the routine doing the check pops always 2 entries at a time. One could equivalently write it as `set(... "GNU" "9.0" "Intel" "19.0")`, but I've found the form above considerably easier to read.

**Review by awvwgk — APPROVED** [review]

**Pull request merged**